include/audio.h

```diff
@@ -83,7 +84,7 @@ class AudioEncoder
 public:
 	// Must have virtual destructor to ensure child class's destructor is called
 	virtual ~AudioEncoder(){};
-	virtual int   Encode(SWORD *in,int inLen,BYTE* out,int outLen)=0;
+	virtual AudioFrame* Encode(const AudioBuffer::const_shared& audioBuffer) = 0;

```

Comment:
can we use a shared pointer better?### 
```suggestion
	virtual AudioFrame::shared Encode(const AudioBuffer::const_shared& audioBuffer) = 0;

```